kuma/wiki/kumascript.py

```diff
@@ -214,8 +214,54 @@ def process_body(response):
     # We defer bleach sanitation of kumascript content all the way
     # through editing, source display, and raw output. But, we still
     # want sanitation, so it finally gets picked up here.
-    return clean_content(response.text)
+    clean_response = clean_content(response.text)
+    # After sanitization perform a final pass to substitute any inlined
+    # interactive example macros
+    return process_inlined_examples(clean_response)
 
+def process_inlined_examples(body):
+    """
+    Manually expand @IExMacro@ declarations
+    """
+    return body.replace("@InlineArrayForEach@", """
+<div class="iex interactive interactive-js">
+<link href="/static/styles/interactive-examples/codemirror-5-31-0.css" rel="stylesheet" />
+<link href="/static/styles/interactive-examples/editor-js.css" rel="stylesheet" />
+
+<script>"use strict";function postToKuma(e){window.parent.postMessage(e,"https://developer.mozilla.org")}postToKuma({markName:"interactive-editor-loading"}),document.addEventListener("readystatechange",function(e){switch(e.target.readyState){case"interactive":postToKuma({markName:"interactive-editor-interactive",measureName:"ie-time-to-interactive",startMark:"interactive-editor-loading",endMark:"interactive-editor-interactive"});break;case"complete":postToKuma({markName:"interactive-editor-complete",measureName:"ie-time-to-complete",startMark:"interactive-editor-loading",endMark:"interactive-editor-complete"})}});</script>
+<section id="static">
+  <pre>
+<code id="static-js">var array1 = ['a', 'b', 'c'];
+
+array1.forEach(function(element) {
+  console.log(element);
+});
+
+// expected output: "a"
+// expected output: "b"
+// expected output: "c"
+</code>
+</pre>
+
+</section>
+
+<section id="live" class="live hidden">
+  <header><h4>JavaScript Demo: Array.forEach()</h4></header>
+  <div id="editor" class="editor"></div>
+
+  <div class="output-container">
+      <div class="buttons-container">
+          <button id="execute" class="button run" type="button">Run &rsaquo;</button>
+          <button id="reset" type="button" class="button">Reset</button>
+      </div>
+      <div id="output" class="output"><code></code></div>
+  </div>
+</section>
+
+<script src="/static/js/interactive-examples/codemirror-5-31-0.js"></script>
+<script src="/static/js/interactive-examples/editor-js.js"></script>
```

Comment:
Would be ideal if we could load the JS files at the bottom along with the other JS. Currently the browser will be blocked when it hits the CSS and JS assets which will not be the case if this is actually baked into Kuma.

It is probably a limitation we need to keep in mind when looking at the numbers. Perhaps we can load the scripts "dynamically" based on the presence of a specific `id` attached to https://github.com/mozilla/kuma/pull/5019/files#diff-bad48985f515ae03ad4b8ad4839f7b2dR227

Perhaps something like:

`<div id="inlined-foreach" class="iex interactive interactive-js">`

then:

```
var inlinedForeach = document.getElementById('inlined-foreach');

if (inlinedForeach) {
    // create script tags and stick into the DOM
}
```

Thoughts @hobinjk

Reply:
I believe my current approach is better than dynamically loading since if we're programmatically creating the script tags I think the browser will be unable to prefetch them. I did a cursory check of the waterfall view and it looks like there is no significant blocking caused by loading the JS files here since it precedes to the bottom-of-page files in parallel
<img width="512" alt="editor-js js inlined doesn't block that much" src="https://user-images.githubusercontent.com/2776089/47817285-a2d4fe00-dd2b-11e8-90d3-00a14b5541c5.png">